django_declarative_apis/machinery/attributes.py

```diff
@@ -378,6 +379,13 @@ def __init__(
         self.retries = retries
         self.retry_exception_filter = retry_exception_filter
 
+        if execute_unless:
+            assert callable(
+                execute_unless
+            ), "execute_unless MUST be an instance method that takes no arguments"
```

Comment:
Should this be a little more strict if it /must/ be an instance method? Something like ensuring `self == execute_unless.__self__`?

Reply:
added an additional check using `inspect` - we can't check for `__self__` because the method isn't bound in this context